Adapt samples to Microsoft/QuantumLibraries#96.

cgranade opened the pull request.

bettinaheim commented on Samples/src/ReversibleLogicSynthesis/ReversibleLogicSynthesis.qs:
```diff
@@ -322,26 +320,26 @@ namespace Microsoft.Quantum.Samples.ReversibleLogicSynthesis {
         }
-        
+
         adjoint invert;
```
Do you want to remove those in favor of a declaration annotation?

bettinaheim commented on Samples/src/ReversibleLogicSynthesis/ReversibleLogicSynthesis.qs:
```diff
@@ -354,33 +352,32 @@ namespace Microsoft.Quantum.Samples.ReversibleLogicSynthesis {
         mutable result = true;
         let nbits = BitSizeI(Length(perm));
-        
         for (i in 0 .. Length(perm) - 1) {
```
Looks like an IndexRange?

bettinaheim commented on Samples/src/ReversibleLogicSynthesis/ReversibleLogicSynthesis.qs:
````diff
@@ -93,18 +93,18 @@ namespace Microsoft.Quantum.Samples.ReversibleLogicSynthesis {
     /// Sequence(23, 29); // [23, 24, 25, 26, 27, 28, 29]
     /// ```
     function Sequence (from : Int, to : Int) : Int[] {
````
We should probably have a good initialization function in the Arrays module...
On one hand we need "conversions" between arrays and ranges (and have a conversation on related type system topics), on the other hand we should also have array initializers taking initialization functions.

cgranade replied:
Very much agreed, let's open an issue for that on QuantumLibraries, then.

bettinaheim commented on Samples/src/ReversibleLogicSynthesis/ReversibleLogicSynthesis.qs:
```diff
@@ -122,17 +122,17 @@ namespace Microsoft.Quantum.Samples.ReversibleLogicSynthesis {
-        
+
         for (i in 0 .. count - 1) {
             set array w/= i <- i;
```
Same here - essentially I'd like to get rid of these, because we should have convenient infrastructure in the array module.

bettinaheim commented on Samples/src/ReversibleLogicSynthesis/ReversibleLogicSynthesis.qs:
```diff
@@ -154,105 +154,103 @@ namespace Microsoft.Quantum.Samples.ReversibleLogicSynthesis {
     function GateMask (controls : Int, targets : Int) : MCMTMask[] {
-        
+
         if (targets != 0) {
```
return a conditional expression instead?

bettinaheim commented on Samples/src/ReversibleLogicSynthesis/ReversibleLogicSynthesis.qs:
```diff
@@ -154,105 +154,103 @@ namespace Microsoft.Quantum.Samples.ReversibleLogicSynthesis {
         let (controls, targets) = gateMask!;
-        
+
         if ((pattern &&& controls) == controls) {
```
return a conditional expression instead?

bettinaheim commented on Samples/src/ReversibleLogicSynthesis/ReversibleLogicSynthesis.qs:
```diff
@@ -154,105 +154,103 @@ namespace Microsoft.Quantum.Samples.ReversibleLogicSynthesis {
-            
+
             for (t in 0 .. Length(tQubits) - 1) {
                 set result = result + [MCTGate(cQubits, tQubits[t])];
```
could be +=

bettinaheim commented on Samples/src/ReversibleLogicSynthesis/ReversibleLogicSynthesis.qs:
```diff
@@ -389,12 +386,11 @@ namespace Microsoft.Quantum.Samples.ReversibleLogicSynthesis {
     /// the first half of the array with the second half of the array.
     operation InnerProduct (qubits : Qubit[]) : Unit {
-        
 		let m = Length(qubits) / 2;
```
there seem to be some tabs left here

bettinaheim commented on Samples/src/ReversibleLogicSynthesis/ReversibleLogicSynthesis.qs:
```diff
@@ -407,17 +403,17 @@ namespace Microsoft.Quantum.Samples.ReversibleLogicSynthesis {
         }
-        
+
         adjoint invert;
```
do you want to get rid of this in favor of a declaration annotation?

cgranade commented:
Thanks for the feedback, @bettinaheim! I've addressed your comments other than those relating to improved array initialization, which is now captured in the new issue Microsoft/QuantumLibraries#104.
